platform/windows/display_server_windows.cpp

```diff
@@ -525,10 +525,30 @@ void DisplayServerWindows::delete_sub_window(WindowID p_window) {
 		wintab_WTClose(windows[p_window].wtctx);
 		windows[p_window].wtctx = 0;
 	}
+
+	HWND next_active = _get_top_window(windows[p_window].hWnd);
+	if (next_active != NULL) {
+		SetActiveWindow(next_active);
+	} else {
+		WARN_PRINT("no _get_top_window found");
+	}
+
 	DestroyWindow(windows[p_window].hWnd);
 	windows.erase(p_window);
 }
 
+HWND DisplayServerWindows::_get_top_window(HWND exclude) {
+	WindowID wd = INVALID_WINDOW_ID;
+	HWND hWnd = NULL;
+	for (Map<WindowID, WindowData>::Element *E = windows.front(); E; E = E->next()) {
+		if (E->get().hWnd != exclude && wd < E->key()) {
+			wd = E->key();
+			hWnd = E->get().hWnd;
+		}
+	}
+	return hWnd;
+}
```

Comment:
You can simplyfy this by using the window_ids instead and use find:

```suggestion
WindowID DisplayServerWindows::_get_top_window(WindowID p_current) {
      ERR_FAIL_COND_V_MSG(p_current == MAIN_WINDOW_ID, INVALID_WINDOW_ID, "The main window has no top window")
      ERR_FAIL_COND_V(p_current == INVALID_WINDOW_ID, INVALID_WINDOW_ID)
      return windows.find(p_current)->prev()->key();
}
```

Reply:
If you use `find()` and then `prev()` you would assume p_current is already the highest WindowID which unfortunately not always true.

For example, a popup window opens the Settings window and then we close the popup window(which is p_current in this case). The Settings windows would have the highest WindowID.

But you are right, this can definitely be simplified. You got me thinking and I just realized this Map is a red-black tree and if I understand the implementation correctly, front() is the lowest WindowID and back() is the highest WindowID. 

 I will compile and test it and make a new commit if it works correctly.

Reply:
If you use back(), you don't even need a new function

Reply:
I compiled the modified code and tested a bit. Everything seems to work as expected. I also removed the new function as suggested and added a few comments.

Please have a look.